### DISCUSSION TITLE
calcKernel missing?

### DISCUSSION THREAD

**Discussion by jrmayeur** [mtex-toolbox/mtex#1466] [Ask Anything] [answered]
Has the calcKernel function been removed in the latest version (5.8.1)? I was in the process of trying optimize the kernel used in computing an ODF from EBSD data according to the "Optimal Kernel Selection" tutorial in the documentation under General Concepts, but the function does not exist in my install.  Has the process for computing/optimizing kernels changed, but is not reflected in updated documentation?

**Comment by kilir** [comment, marked as answer]
Hi,
it shoulb be there.
https://github.com/mtex-toolbox/mtex/blob/develop/geometry/@orientation/calcKernel
Cheers,
Rüdiger

**Reply by jrmayeur** [reply]
Hi,
It is in fact there in my installation, but not recognized/callable from the command line. But that is not your problem to solve.
Thanks,
Jason

Update:  The issue was that I hadn't defined an orientation variable (e.g. ori = ebsd.orientations) prior to trying to call the method. I suppose that the orientation class methods are only available at the command line only if there is an available instance in memory.  I was just trying to use calcKernel(ebsd.orientations) directly.  Sorry, it's been quite a long time since I've used MTEX. A lot has changed and I'm a bit rusty.

**Reply by kilir** [reply]
Hi,
how do you call the function? Input with a list of orientations?
Cheers,
Rüdiger
